src/commands.rs

```diff
@@ -1102,6 +1106,27 @@ struct GlobalArgs {
     at_operation: String,
 }
 
+impl GlobalArgs {
+    fn merge_from(&mut self, other: &GlobalArgs) {
+        // Destructure `other`, so we're forced to update this code if new fields are
+        // added in the future.
+        let GlobalArgs {
+            repository,
+            no_commit_working_copy,
+            at_operation,
+        } = other;
+        if self.repository.is_none() {
```

Comment:
`if repository.is_some() {` for consistency?

Other than that, the code looks good to me.

Reply:
Oops, good catch! Fixed.